.travis.yml

```diff
@@ -8,6 +8,12 @@ services:
 python:
   - "2.7"
 
+env:
+  # run flake8 check on all python files in the project
+  - TEST_CMD="ci/flake8_diff.sh"
+  # run the full testsuite
+  - TEST_CMD="docker exec sardana-test sardanatestsuite"
+
```

Comment:
Using stages would mean changing the approach completely,  since they run sequentially and  a stage failure makes the whole pipe to break. That would mean loosing information in case of a failure - we will only learn about the state of one of the checks.

Apart of the option that you suggest (conditional install) we could also revert just the commit c592f92, accepting the two others and run one job only. 

I will investigate a little bit more on that..